drivers/timer/arcv2_timer0.c

```diff
@@ -272,3 +341,17 @@ void sys_clock_disable(void)
 
 	irq_disable(IRQ_TIMER0);
 }
+
+
+#ifdef CONFIG_SMP
+void smp_timer_init(void)
+{
+	/* ensure slave core's internal timer doesn't work */
+	timer0_control_register_set(0);
+	timer0_count_register_set(0);
+	timer0_limit_register_set(0);
+
+	z_irq_priority_set(IRQ_TIMER0, CONFIG_ARCV2_TIMER_IRQ_PRIORITY, 0);
+	irq_enable(IRQ_TIMER0);
```

Comment:
Comment above says "ensure slave core's internal timer doesn't work" still we enable Timer0 interrupt here even though in the peripheral (Timer0) we just disabled interrupt generation by writing 0 to its `control` register. That doesn't look good.

Reply:
the comment needs to be updated.   It's possible that each core's local timer interrupt is raised.